tideways_xhprof.c

```diff
@@ -219,7 +219,7 @@ ZEND_DLEXPORT void tideways_xhprof_execute_internal(zend_execute_data *execute_d
     int is_profiling = 1;
 
     if (!TXRG(enabled) || ((TXRG(flags) & TIDEWAYS_XHPROF_FLAGS_NO_BUILTINS) > 0) ||
-        zend_string_equals_literal(execute_data->func->common.function_name, "tnprof_function_begin")) {
+        (strncmp(ZSTR_VAL(execute_data->func->common.function_name), "tnprof_", 7) == 0)) {
```

Comment:
You need to check ZSTR_LEN of execute_data->func->common.function_name. If it is less than 7 characters, you will cause a buffer overrun.

Reply:
My mistake, strncmp checks for a NULL character.